tests/rptest/tests/transactions_test.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+
+import time
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_consumer import RpkConsumer
+import confluent_kafka as ck
+from rptest.clients.kafka_cat import KafkaCat
+
+import subprocess
+
+
+class TransactionsTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3),
+              TopicSpec(partition_count=1, replication_factor=3))
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "transaction_coordinator_replication": 3,
+            "id_allocator_replication": 3,
+            "enable_leader_balancer": False,
+            "enable_auto_rebalance_on_node_add": False
+        }
+
+        super(TransactionsTest, self).__init__(test_context=test_context,
+                                               extra_rp_conf=extra_rp_conf)
+
+        self.input_t = self.topics[0]
+        self.output_t = self.topics[1]
+        self.max_records = 100
+
+    def generate_data(self, topic, num_records):
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+        })
+
+        for i in range(num_records):
+            producer.produce(topic.name, str(i), str(i), 0)
+
+        producer.flush()
+
+    def check_consumer(self, consumer, use_timeout_for_consume=True):
+        def consume_records():
+            for_one_consume = 10
+            timeout = 2
+            if use_timeout_for_consume:
+                self.records = consumer.consume(for_one_consume, timeout)
```

Comment:
consumer.consume doesn't guarantee to return exact `num_messages` (`for_one_consume` in our case). The [doc](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.consume) says it's maximum:

> num_messages (int) – The maximum number of messages to return (default: 1).

but nothing prevent to to return anything less. In that case the later call overwrite earlier call. Think of the consume as a cursor which returns a random amount of data and shifting its position by the number of records it returns.